### DISCUSSION TITLE
Unexplainable difference in results from reproject compared to what it should be.

### DISCUSSION THREAD

**Discussion by avhuf** [DotSpatial/DotSpatial#1557] [Q&A]
Hi,
I need to transform coordinates from WGS84 (EPSG:4326) to Belgian Lambert 72 (EPSG:31370).

I'm using the following testcode (C#):

```
using DotSpatial.Projections;

class Program
{
    static void Main()
    {
        // Define source and target projections
        ProjectionInfo wgs84 = KnownCoordinateSystems.Geographic.World.WGS1984;
        ProjectionInfo lambert72 = KnownCoordinateSystems.Projected.NationalGrids.BelgeLambert1972;

        // Coordinates to transform (longitude, latitude)
        double[] xy = { 3.679961797, 51.084861098 };
        double[] z = { 0 }; // Z-coordinate if needed (optional)

        // Transform from WGS84 to Lambert 72
        Reproject.ReprojectPoints(xy, z, wgs84, lambert72, 0, 1);

        // Output transformed coordinates
        Console.WriteLine($"X: {xy[0]}, Y: {xy[1]}");
    }
}
```

Now, with the coordinates given in the code, the result SHOULD be: (X: 101737.57593892881, Y: 197417.13309476897).
But when you run the code, the results ARE: X: 101824,585983408, Y: 197356,927404788

Even when I replace "KnownCoordinateSystems.Projected.NationalGrids.BelgeLambert1972" by the EPSG31370 definition this still gives the wrong result.

The same transformation in pyproj  for instance gives the correct result. What could be the reason here?

Thx

Peter

**Comment by sindizzy** [comment]
Have you looked at the actual projection info? Maybe some of the parameters are different.

**Reply by avhuf** [reply]
I have looked around at the different EPSG on spatialreference.org to see whether I had more success than with KnownCoordinateSystems.Projected.NationalGrids.BelgeLambert1972. There are many versions of Belge Lambert 1972. It's a mess.

After testing more, the configuration with which I was successful is:
proj4_31370 = "+x_0=150000.013 +y_0=5400088.438 +lat_0=90 +lon_0=4.3674866666666663 +lat_1=51.166667233333328 +lat_2=49.8333339 +proj=lcc +towgs84=-106.869,52.2978,-103.724,0.3366,-0.457,1.8422,-1.27469999999796 +ellps=intl +no_defs";

This gives the same result as you can get through for instance:
1) https://www.tool-online.com/en/coordinate-converter.php# (using the one with CORRECTED parameters, not the one that is EPSG 31370).
2) the conversion tool on the belgian national geographical institute. (https://ngi.be/online-resources/cconvert/)

My conclusion is that the **KnownCoordinateSystems.Projected.NationalGrids.BelgeLambert1972** included with Dotspatial.projections is not the correct version that should officially be used for Belge Lambert 1972.

Regards
Peter

**Reply by sindizzy** [reply]
I believe they get those from the PROJ4 library. Maybe you can see what is defined there as well.
